### PR TITLE
fix: safely handle None in prepare_response (fixes #17891)

### LINKED ISSUE #17891: TypeError: 'NoneType' object is not iterable
There is an issue when the webhook doesn't respond, and the `paymentGatewayInitialize` mutation receives the following variables in the `prepare_response` method:

```
payment_gateways_input = [PaymentGatewayData(app_identifier='saleor.app.payment.stripe', data=None, error=None)]
payment_gateways_response = None
```

This results in the mutation failing with the following exception:

```
TypeError: 'NoneType' object is not iterable
(1 additional frame(s) were not displayed)
...
  File "saleor/graphql/core/mutations.py", line 521, in mutate
    response = cls.perform_mutation(root, info, **data)
  File "saleor/graphql/payment/mutations/transaction/payment_gateway_initialize.py", line 149, in perform_mutation
    gateway_configs=cls.prepare_response(payment_gateways_data, response_data),
  File "saleor/graphql/payment/mutations/transaction/payment_gateway_initialize.py", line 81, in prepare_response
    gateway.app_identifier: gateway for gateway in payment_gateways_response

A query failed unexpectedly
```

First seen in `3.22.0-a.0`.

Internal Sentry issue: [SALEOR-CORE-6GE](https://saleor.sentry.io/issues/6717087207/?referrer=github_integration)

### PR DESCRIPTION
**Pull request by Ruhi14**
I want to merge this change because...

This pull request fixes [#17891](https://github.com/saleor/saleor/issues/17891):
When the payment gateway webhook does not respond, the PaymentGatewayInitialize mutation receives None for payment_gateways_response. Previously, this caused a runtime error (TypeError: 'NoneType' object is not iterable) in the prepare_response method, breaking the mutation execution flow.

With this change, prepare_response now treats None as an empty list, which prevents errors and ensures payment gateway initialization continues robustly, even if a webhook fails to respond.

Relevant Issue: #17891

Impact
 Backend bugfix (no DB, API, or schema changes)

 New migrations

 New/Updated API fields or mutations

 Deprecated API fields or mutations

 Removed API types, fields, or mutations

Docs
 No documentation update required
(This fix concerns backend robustness only—no externally visible API or docs change.)

Pull Request Checklist
 My code follows existing style and conventions.

 I have manually verified the bug and bugfix locally.

 I have added or updated tests (not added; see note below).

 I have updated the documentation (not required for this change).

Additional Notes
No migrations, schema changes, or API updates are included.

This PR improves error handling for payment gateway integration and prevents possible mutations from failing if an external payment app is down or unresponsive.

Why no test? I was unable to include a full automated test in this PR due to environment constraints. I recommend maintainers add/extend a test for this edge case as appropriate.

Please let me know if you need help with a test, or you would like me to contribute further.

Thank you for reviewing!

### CODE REVIEW

**Comment by korycins** [comment]
Hey @Ruhi14  please see my comment from: https://github.com/saleor/saleor/pull/17892#issuecomment-3021977461

File: saleor/graphql/payment/mutations/transaction/payment_gateway_initialize.py [hunk 1]
```diff
@@ -77,6 +77,9 @@ def prepare_response(
         payment_gateways_response: list[PaymentGatewayData],
     ) -> list[PaymentGatewayConfig]:
         response = []
+        # ✅ Fix: Handle case where payment_gateways_response is None
+        if not payment_gateways_response:
+            payment_gateways_response = []
```
Comment (patrys) [inline saleor/graphql/payment/mutations/transaction/payment_gateway_initialize.py:82]: This function cannot receive `None` as the argument type is `list`. If there is a bug, it's not in this function but in one of the callers.

Comment (Ruhi14) [reply saleor/graphql/payment/mutations/transaction/payment_gateway_initialize.py:82]: Hi @patrys, thanks for the clarification!

You're absolutely right — prepare_response is annotated to receive a List, so None should never reach this function. After reviewing the flow again, it looks like the root cause is in the caller where payment_gateways_response can be None if the webhook does not return a response.

I'll move the None check and fallback to an empty list to the calling function (likely in resolve_payment_gateway_initialize) and update the PR accordingly.

Thanks again for pointing it out! 🙌

**Review by Ruhi14 — COMMENTED** [review]
I've updated the fix to handle None in the caller function instead of inside prepare_response, as suggested. Let me know if any other changes are needed!

**Comment by korycins** [comment]
Hey @Ruhi14, I don't think that this is proper way of fixing it. This is my comment that I've placed in #17892 , as it is also valid in case of your PR. 

> I don't think that this is the proper place of fixing it. The whole typing on:
payment_gateway_initialize-> PluginsManager -> WebhookPlugin expects return type: list["PaymentGatewayData"].
If we somehow returns None to paymentGatewayInitialize, then it is issue in WebhookPlugin, and Webhook plugin should align to the expected return type.

 Also please remember that in case of any changes in the code, we also need some additional tests to confirm that applied fixes working as expected.

**Comment by wcislo-saleor** [comment]
If you intend to continue working on this please create a new pull request.
